prog/dftb+/lib_type/dynneighlist.F90

```diff
@@ -0,0 +1,244 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2017  DFTB+ developers group                                                      !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+
+!> Implements dynamic neighbor list with iterator.
+!>
+!> The dynamic neighbor list does not store the entire neighbor list, but creates it on the fly
+!> allowing for a low memory footprint for large neighbor list (at the cost of speed).
+!>
+module dynneighlist
+  use accuracy
+  use assert
+  use latpointiter
+  use message
+  implicit none
+  private
+
+  public :: TDynNeighList, TDynNeighList_init
+  public :: TNeighIterator, TNeighIterator_init
+
+
+  !> Dynamic neighbor list
+  type :: TDynNeighList
+    private
```

Comment:
Even though these are private, for future new developers, would be nice if the variables were commented.

Reply:
Absolutely correct, I did that in both files now.